pkg/operator/k8sutil/node.go

```diff
@@ -329,33 +329,41 @@ func RookNodesMatchingKubernetesNodes(rookStorage rookv1.StorageScopeSpec, kuber
 }
 
 // ExtractTopologyFromLabels extracts rook topology from labels and returns a map from topology type to value
-func ExtractTopologyFromLabels(labels map[string]string, additionalTopologyIDs []string) map[string]string {
+func ExtractTopologyFromLabels(labels map[string]string, additionalTopologyIDs []string) (map[string]string, string) {
 	topology := make(map[string]string)
 
+	// The topology affinity for the osd is the lowest topology label found in the hierarchy,
+	// not including the host name
+	var topologyAffinity string
```

Comment:
This references OSD, but this is operator-independent code.

I think it's reasonable to return the lowest topology label in the hierarchy from this function, but maybe it should just return the map key and allow the calling function to decide what to do with the information. In Ceph's case, we want to make a `key=value` pair, but if we still assume other backends might reuse this code, let's not pigeonhole them into a use-case.

```go
topology, lowest := ExtractTopologyFromLabels(mylabels, additionalTopologyIDs)
topologyAffinity := fmt.Sprintf("%s=%s, lowest, topology[lowest])
```

Reply:
> This references OSD, but this is operator-independent code.

This topology isn't used by any other storage provider, so until it is needed elsewhere I think it best to move this back into a ceph-specific package to avoid this confusion.

> I think it's reasonable to return the lowest topology label in the hierarchy from this function, but maybe it should just return the map key and allow the calling function to decide what to do with the information. In Ceph's case, we want to make a key=value pair, but if we still assume other backends might reuse this code, let's not pigeonhole them into a use-case.

The key of the `topology` map actually doesn't necessarily map the `lowest` value. The keys of the `topology` map are just the simplified names such as `zone` instead of the full label name.